hedera-mirror-grpc/src/main/java/com/hedera/mirror/grpc/controller/ConsensusController.java

```
@@ -104,4 +121,13 @@ private Throwable unknownError(Throwable t) {
         log.error("Unknown error subscribing to topic", t);
         return Status.INTERNAL.augmentDescription(t.getMessage()).asRuntimeException();
     }
+
+    private boolean validEndTimeStamp(Timestamp endTimeStamp) {
+        if (endTimeStamp == null) {
+            return false;
+        }
+
+        return (endTimeStamp.getSeconds() >= 0 && endTimeStamp.getSeconds() <= LONG_MAX_SECONDS) &&
```

This should fail if endTimestamp is negative instead of assuming they want to never end

---

Addressed